Cloudformation

richford opened the pull request:
This PR removes the cloudknot objects that represent IAM roles, VPCs, security groups, job definitions, compute environments, and job queues. In their place, `Pars` and `Knot` now use AWS CloudFormation templates to create resource stacks.

richford commented:
Whew! Net change of -4392 lines. <kbd>delete</kbd> is the most productive key on my keyboard!
@arokem, when you get back from vacation, please take this for a test drive. I want to make sure I didn't break anything.

arokem commented:
Wow! That's great. Do you understand the CI failure?

arokem commented:
When I try to run the example in the front page, I get: 

```
ClientError: An error occurred (ValidationError) when calling the CreateStack operation: 1 validation error detected: Value 'random_mv_prod-pars' at 'stackName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z][-a-zA-Z0-9]*
```

arokem commented:
This is circumvented by changing the name to something else (e.g., `rndmvprd`)

richford commented:
Good catch, I guess underscores are not allowed in stack names. I added regex verification to the `NamedObject` base class and automatically converted underscores to hyphens. Also updated the docs accordingly.

Not sure about the travis error since I explicitly put that statement in a try/except block and passed in the except block.

richford commented:
Fixed the travis errors.
